src/main/java/it/polimi/telco/model/Product.java

```diff
@@ -0,0 +1,55 @@
+package it.polimi.telco.model;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+
+@Entity
+@Table(name = "product", schema = "telco")
+@NamedQuery(name = "Product.getProductById", query = "SELECT p FROM Product p  WHERE p.id = ?1 ")
+public class Product {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @NotNull
+    private String name;
+
+    @NotNull
+    private double monthlyFee;
+
+    @NotNull
+    @OneToOne
+    private ValidityPeriod validityPeriod;
```

Comment:
_"The validity period of an optional product is the same as the validity period that the user has chosen for the service package. An optional product has a name and a monthly fee independent of the validity period duration"_

I think that an optional product is not associated with a validity period (it inherits it when is associated with a subscription)